zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/UncontrolledStreamClock.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.stream.impl;
+
+import io.camunda.zeebe.stream.api.StreamClock;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.InstantSource;
+import java.time.ZoneId;
+import java.util.Objects;
+
+public final class UncontrolledStreamClock implements StreamClock {
```

Comment:
:thought_balloon: Likely if we did not implement `InstantSource` but our own type like `EpochClock`, then we wouldn't need an implementation here but could return a constant lambda, e.g. `() -> InstantSource.system().millis()`.

Reply:
Actually this implementation is already not used, maybe we can just remove it along with the interface `StreamClock`? I don't think we have a usecase where the clock should _not_ be controllable.

Reply:
Th only risk is in the future if we want an immutable one :shrug: I could imagine that we will have a feature flag for this, so the question is do we want to make it completely impossible, or just prevent commands from modifying the clock? I would lean towards the safer/cautious approach, but in the end with good test coverage it's likely fine either way.

Reply:
Okay then let's keep it for now, maybe we can use it to disable the entire feature :+1: